app/helpers/search_helper.rb

```diff
@@ -85,6 +85,20 @@ def ordering
     end
   end
 
+  # Transforms controller params from get request into query args for a search
+  # @return [Hash]
+  def get_query_args
```

Comment:
You may want to consider using a client-side excel export solution for this. Check out something like [excel-export](https://www.npmjs.com/package/excel-export) or [react-csv](https://www.npmjs.com/package/react-csv).

Another option would be store the export file in S3 and make S3 the download source. We may be able to use something like [bucketeer](https://devcenter.heroku.com/articles/bucketeer) to easily integrate S3 sources into Heroku.

I've actually got experience implementing both of these approaches (albeit using CoffeeScript and Sinatra instead of React and Rails) for document sets at different scales, and they do work.

One thing to consider about putting a file download within the context of a request is the risk of a timeout. Transforming a sufficiently large data set may cause data to process for longer than the default 30 second timeout that is customary for a request to Rails on Heroku. We should be able to solve this problem by performing these calculations client side through a paginated series of requests -- using the existing search API available in the backend.